illumos-utils/src/zpool.rs

```diff
@@ -284,12 +284,39 @@ pub enum ZpoolKind {
 ///
 /// This expects that the format will be: `ox{i,p}_<UUID>` - we parse the prefix
 /// when reading the structure, and validate that the UUID can be utilized.
-#[derive(Clone, Debug, Hash, PartialEq, Eq, JsonSchema)]
+#[derive(Clone, Debug, Hash, PartialEq, Eq)]
 pub struct ZpoolName {
     id: Uuid,
     kind: ZpoolKind,
 }
 
+/// Custom JsonSchema implementation to encode the constraints on Name.
+impl JsonSchema for ZpoolName {
```

Comment:
Why am I doing this?

This type has a custom serialization / deserialization method, to encode it as "ox{i,p}_UUID". That's how it's currently represented in e.g. TOML config files.

IMO this format is kinda nice, because it exactly matches the name a user can see if they run `zpool list`. At the same time, the in-memory structure is nice, because it lets us do handy things like "match on `kind`".

Well, this works when serializing/deserializing to TOML, but sorta broke when using `JsonSchema` and progenitor -- in that case, it was serialized as `{"id":..., "kind":....}`, and couldn't be deserialized from that format.

By using this custom `JsonSchema` impl, this type is always just serialized "as a single string".

Reply:
It might be useful to have a regular expression in this. That can help progenitor create better error messages, for example.

Reply:
Added in ae93a3ece49a63da7a352edbfd7d111dd086953f , with tests!